### PR TITLE
Minor documentation improvements

### PR DESCRIPTION
**Pull request by nmnobre**
Hi,

Just a collection of very minor typo fixes and rewordings.

Cheers,
-Nuno

### CODE REVIEW

File: src/maxwell-notes.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ With electric current density, ${\bf J}$, magnetic current density,
 ${\bf M}$, electric conductivity, $\overline{\sigma}$, magnetic
 conductivity, $\overline{\sigma}_M$, and electric charge density,
 $\rho$. We will sometimes refer to these equations by the names Ampère's
-Law, Faraday's Law, Guass' Law, and the Transversality Condition
+Law, Faraday's Law, Gauss's Law, and the Transversality Condition
```
Comment (tzanio) [inline src/maxwell-notes.md:14]: We need to fix the `Guass` misspelling, but I believe either `Gauss' Law` or `Gauss's Law` is fine here.

I personally prefer `Gauss' Law`.

File: src/building.md [hunk 2]
```diff
@@ -233,9 +233,9 @@ To enable package testing during the build process, use instead:
...
 spack install -v --test=all mfem
 ```
-If you don't have Spack, you can download it and install MFEM with the following commands:
+If you don't have Spack, you can download it and setup your environment with the
+following commands, after which you can install MFEM as above:
 ```sh
 git clone https://github.com/spack/spack.git
-cd spack
-./bin/spack install -v mfem
+. spack/share/spack/setup-env.sh
```
Comment (tzanio) [inline src/building.md:240]: This changes the meaning -- the intention was for people to be able to copy and paste and get MFEM built with Spack from scratch.

Maybe that's not a good idea, and we should just tell them where to get Spack from...

I don't think we should include the `. spack/share/spack/setup-env.sh` line though -- it doesn't work for all shells (e.g. I use `tcsh` and it doesn't work for me), and it is not necessary: for a quick build you can just use `./bin/spack`.

**Comment by tzanio** [comment]
Thanks for these fixes @nmnobre, they are very welcome!

I made two comments above, what do you think?

File: src/maxwell-notes.md (hunk 1 above)
Comment (nmnobre) [reply src/maxwell-notes.md:14]: Hi Tzanio,

I'm happy to revert the change, but note that every other document on the website refers to it as `Gauss's Law`.
I'll let you make the final call.

File: src/building.md (hunk 2 above)
Comment (tzanio) [reply src/building.md:240]: > Shall I just drop the commit and force push?

Lets do that

Comment (nmnobre) [reply src/building.md]: Done.

**Review by tzanio — APPROVED** [review]
Thanks @nmnobre !

**Pull request merged**